library/src/scala/quoted/Quotes.scala

```diff
@@ -4057,6 +4013,59 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
 
     end TreeMap
 
+    /** Type class used in `show` methods to provide customizable `String` representations */
+    trait Printer[T]:
+      /** Show the arguments as a `String` */
+      def show(x: T): String
```

Comment:
With Printer being a member of the Reflection trait, it seems very tricky for users to write their own custom printer that works with any Reflection context. Is there an example of how to do that? If it's not possible maybe we can come up with a better API here.

Reply:
There are examples of how to use the Tree map that is quite similar. I will add an example for printer.

Reply:
That would be nice.